File: dsa/3-arrays/challenges/03-integer-reversal/integerReversal.js
function reverseNumber(number) {
    if (typeof number != "number") throw new Error('Input must be a number')
    const sign = Math.sign(number)
    const reversed = Number(Math.abs(number).toString().split('').reverse().join(''))
    console.log(reversed * sign)
}

reverseNumber(-123456)

File: dsa/3-arrays/challenges/03-integer-reversal/integer-reversal.py
def reverse_number(number):
    if not isinstance(number, (int, float)):
        raise ValueError("Input must be a number")

    sign = -1 if number < 0 else 1
    if isinstance(number, int):
        reversed_int_str = str(abs(int(number)))[::-1]
        print(int(reversed_int_str) * sign)
    elif isinstance(number, float):
        reversed_float_str = str(abs(float(number)))[::-1]
        print(float(reversed_float_str) * sign)


reverse_number(-123.23)
